great_expectations/execution_engine/sqlalchemy_execution_engine.py

```diff
@@ -1058,6 +1060,9 @@ def resolve_metric_bundle(
                     )
 
                 logger.debug(f"Attempting query {str(sa_query_object)}")
+
+                if sqlalchemy_Engine and isinstance(self.engine, sqlalchemy_Engine):
+                    self.engine = self.engine.connect()
```

Comment:
This line bug me a bit. I wonder if it causes some inconsistencies, since `self.engine` can be either an `Engine` or`Connection`

Reply:
It makes sense I think. where would the logic for the "choice" between Engine and Connection live? As the code currently is, I think it live at the `SqlAlchemyExecutionEngine` (ie. here) + tests that deal with `SqlAlchemyBatchData`. 

I wonder if a helper method could be used here.

Reply:
Yeah I agree. Something like the `get_connection()` that yields a connection that we discussed previously. And returns the same connection if we are using a specific dialect that requires the connection to stay open to keep temp tables alive.